torch/_meta_registrations.py

```diff
@@ -7583,6 +7583,20 @@ def _constant_pad_nd_meta(input, pad, value=0):
         f"{l_inp} dimensions.",
     )
 
+    if all(isinstance(p, utils.IntWithoutSymInt) and p <= 0 for p in pad):
```

Comment:
Why doesn't this logic work with symint?

Reply:
Because we can't do a branching based on the symint value in the meta function AFAIK.